worker/wallet.js

```diff
@@ -284,3 +286,73 @@ export async function checkPendingWithdrawals (args) {
     }
   }
 }
+
+export async function checkWallet ({ data: { userId }, models }) {
+  // This function runs when a row was inserted, updated or deleted from the wallet table
+  // and is given the userId of the user that the wallet belongs (or belonged in the case of deletion) to
+  const wallets = await models.wallet.findMany({
+    where: {
+      userId,
+      enabled: true
+    },
+    include: {
+      vaultEntries: true
+    }
+  })
+
+  const { hasRecvWallet: oldHasRecvWallet, hasSendWallet: oldHasSendWallet } = await models.user.findUnique({ where: { id: userId } })
+
+  const newHasRecvWallet = wallets.some(({ type, wallet }) => canReceive({ def: getWalletByType(type), config: wallet }))
+  const newHasSendWallet = wallets.some(({ vaultEntries }) => vaultEntries.length > 0)
+
+  await models.user.update({
+    where: { id: userId },
+    data: {
+      hasRecvWallet: newHasRecvWallet,
+      hasSendWallet: newHasSendWallet
+    }
+  })
+
+  const startStreak = async (type) => {
+    const streak = await models.streak.create({
+      data: { userId, type, startedAt: new Date() }
+    })
+    return streak.id
+  }
+
+  const endStreak = async (type) => {
+    const [streak] = await models.$queryRaw`
+      UPDATE "Streak"
+      SET "endedAt" = now(), updated_at = now()
+      WHERE "userId" = ${userId}
+      AND "type" = ${type}::"StreakType"
+      AND "endedAt" IS NULL
+      RETURNING "id"
+    `
+    return streak?.id
```

Comment:
in case we ever start to delete streaks, this optional chaining plus `if (streakId)` will handle that

we don't do that currently in prod afaik, but during development, I tended to delete existing streaks for testing purposes so this was useful